parsec/backend/pki.py

```diff
@@ -214,117 +235,106 @@ async def api_pki_enrollment_info(
+            PkiEnrollmentAnswerPayload.load(msg.accept_payload)
 
         except DataError as exc:
-            return {"status": "invalid_payload_data", "reason": str(exc)}
+            return PkiEnrollmentAcceptRepInvalidPayloadData(str(exc))
 
         try:
             user, first_device = validate_new_user_certificates(
                 expected_author=client_ctx.device_id,
                 author_verify_key=client_ctx.verify_key,
-                device_certificate=msg["device_certificate"],
-                user_certificate=msg["user_certificate"],
-                redacted_user_certificate=msg["redacted_user_certificate"],
-                redacted_device_certificate=msg["redacted_device_certificate"],
+                device_certificate=msg.device_certificate,
+                user_certificate=msg.user_certificate,
+                redacted_user_certificate=msg.redacted_user_certificate,
+                redacted_device_certificate=msg.redacted_device_certificate,
             )
 
         except CertificateValidationError as exc:
-            return pki_enrollment_accept_serializer.rep_dump(
-                {"status": exc.status, "reason": exc.reason}
-            )
+            return PkiEnrollmentAcceptRepInvalidCertification(exc.reason)
 
         try:
             await self.accept(
                 organization_id=client_ctx.organization_id,
-                enrollment_id=msg["enrollment_id"],
-                accepter_der_x509_certificate=msg["accepter_der_x509_certificate"],
-                accept_payload_signature=msg["accept_payload_signature"],
-                accept_payload=msg["accept_payload"],
+                enrollment_id=msg.enrollment_id,
+                accepter_der_x509_certificate=msg.accepter_der_x509_certificate,
+                accept_payload_signature=msg.accept_payload_signature,
+                accept_payload=msg.accept_payload,
                 accepted_on=DateTime.now(),
                 user=user,
                 first_device=first_device,
             )
-            rep = {"status": "ok"}
+            return PkiEnrollmentAcceptRepOk()
 
         except PkiEnrollmentNotFoundError as exc:
-            rep = {"status": "not_found", "reason": str(exc)}
+            return PkiEnrollmentAcceptRepNotFound(str(exc))
 
         except PkiEnrollmentNoLongerAvailableError as exc:
-            rep = {"status": "no_longer_available", "reason": str(exc)}
+            return PkiEnrollmentAcceptRepNoLongerAvailable(str(exc))
 
         except PkiEnrollmentAlreadyExistError as exc:
-            rep = {"status": "already_exists", "reason": str(exc)}
+            return PkiEnrollmentAcceptRepAlreadyExists(str(exc))
 
         except PkiEnrollmentActiveUsersLimitReached:
-            rep = {"status": "active_users_limit_reached"}
-
-        return pki_enrollment_accept_serializer.rep_dump(rep)
+            return PkiEnrollmentAcceptRepActiveUsersLimitReached(None)
```

Comment:
Does this Rep have a reason field ?

Reply:
There's not such fields in the json schema